arangod/Statistics/ServerStatistics.h

```diff
@@ -24,18 +24,22 @@
 #ifndef ARANGOD_STATISTICS_SERVER_STATISTICS_H
 #define ARANGOD_STATISTICS_SERVER_STATISTICS_H 1
 
+#include <atomic>
+#include <memory>
+
 class ServerStatistics {
  public:
-  static ServerStatistics statistics();
+  friend std::unique_ptr<ServerStatistics> std::make_unique<ServerStatistics>(double&);
```

Comment:
Looks overly complicated. Why do we need this at all?

Reply:
Is using a `unique_ptr` worse than a `delete` in the `stop()` function of the feature?

Reply:
unique_ptrs aren't bad. friend declarations are. why do we need a friend declaration?

Reply:
I am still not convinced we need a friend declaration.
We seem to have a global object anyway, so why don't we make the ctor public, remove the unique ptr and create the global object in the translation unit before main()? 
And then just forbid copying/assigning of ServerStatistics and it should be fine.
That would make things even less complicated, doesn't require a friend declaration etc.

Reply:
Should I drop the const for the start_time? If we have a problem with the initialization order now we get a nullptr access and a segfault. If we change as you proposed that kind of error becomes less obvious. It probably does not really matter as it will only be a fraction's fraction of the server uptime.

Reply:
I have added the changes as requested. In terms of line numbers is has about the same complexity.